### ISSUE TITLE
Sage Intacct - Taxes are not imported for sage intacct integration

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53935]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.74-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+acc1te2@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Prerequisite
Have a workspace connected to Sage Intacct
1. Navigate to the workspace connected to Sage Intacct 
2. Navigate to Accounting > Import > Tax > Enable Tax
3. Wait for the sync to finish

## Expected Result:
Taxes are imported and displayed on the menu

## Actual Result:
Taxes are not imported to the workspace after the sync is finished

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/61295963-2994-47a6-88f2-dfd73ab04f5c

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by danieldoglas** [comment]
@yuwenmemon can you maybe advise on the expected behavior here and if the proposed changes looks good?

**Comment by yuwenmemon** [comment]
The key is `syncTax` - not 
```diff
+                tax: {
+                    trackingEnabled: enabled,
+                },
```

**Comment by yuwenmemon** [comment]
@dsilva Just look up the API and follow the code in the back-end and you should be able to make an assessment on the proposal: https://github.com/Expensify/Web-Expensify/blob/2cb137fa83983e634c1498a18f6bb82519e9769e/api.php#L2598-L2604

Lemme know if you have any questions!

**Comment by danieldoglas** [comment]
I'll assign it to you then since I have competing priorities now. Thanks for offering.

**Comment by yuwenmemon** [comment]
I guess I'm confused then... why do we have `trackingEnabled` in the FE? It should be set to mirror the BE ideally.

**Comment by garrettmknight** [comment]
@allgandalf can you respond to @yuwenmemon's question [here](https://github.com/Expensify/App/issues/53935#issuecomment-2560536965)?

**Comment by yuwenmemon** [comment]
This is the structure of the config object for Sage Intacct:
```
    "config": {
        "importEmployees": false,
        "isConfigured": true,
        "approvalMode": null,
        "isAutoSyncEnabled": false,
        "mappings": {
            "projects": "TAG",
            "classes": "TAG",
            "locations": "TAG",
            "departments": "TAG",
            "customers": "TAG",
            "dimensions": []
        },
        "credentials": {
            "companyID": "Expensicorp-Skyline",
            "userID": "emma"
        },
        "autoSync": {
            "jobID": null,
            "enabled": false
        },
        "tax": {
            "syncTax": true,
            "taxSolutionID": "United Kingdom - VAT"
        },
        "export": {
            "nonReimbursableVendor": "20003",
            "exportDate": "LAST_EXPENSE",
            "exporter": "exporter@domain.com",
            "reimbursable": "VENDOR_BILL",
            "nonReimbursable": "CREDIT_CARD_CHARGE",
            "nonReimbursableCreditCardChargeDefaultVendor": null,
            "reimbursableExpenseReportDefaultVendor": null,
            "nonReimbursableAccount": "CC1"
        },
        "sync": {
            "syncReimbursedReports": false,
            "reimbursementAccountID": null
        },
        "entity": "400"
    }
```

```
        "tax": {
            "syncTax": true,
            "taxSolutionID": "United Kingdom - VAT"
        },
```

We never use the term `trackingEnabled`

**Comment by yuwenmemon** [comment]
Sounds good - marking internal

**Comment by yuwenmemon** [comment]
Will look next week.

**Comment by garrettmknight** [comment]
@yuwenmemon did you get a chance to look at this one?

**Comment by yuwenmemon** [comment]
Looking at this today!

**Comment by yuwenmemon** [comment]
So this is probably a front-end error. It looks like we missed adding consideration for Intacct's `taxSolutionIDOptions` which need to be chosen from when turning on tax.

First, we'll need to get an Intacct account that actually has tax enabled. @francoisl I know you implemented this part of the integration back in the day. Do you know what Intacct test account is enabled for tax configuration that I could test with?

**Comment by francoisl** [comment]
Yeah try "Sage Intacct with Tax - API account creds" from 1Password

**Comment by yuwenmemon** [comment]
D'oh. Thank you 😄

**Comment by yuwenmemon** [comment]
Okay, here's what we want to build. Thankfully everything we need is present already and being imported into connections.

For starters, we need to check if the connection's `data.taxSolutionIDs` property is filled. If it's not, we should not show the tax selector at all:

![Image](https://github.com/user-attachments/assets/88434edb-fe90-432b-b6c8-9b312627a49b)

Then, if `data.taxSolutionIDs` is filled, it will contain data like the following:

![Image](https://github.com/user-attachments/assets/0b2c3e9d-a09b-4c15-b0c6-e379e7e1193c)

In this case, we should show the Tax selector but it should look more like the other dimension selectors in the import tab. Here is a mock up:

![Image](https://github.com/user-attachments/assets/ae9925d8-31ee-4fb9-9e97-ac2a8f957dce)

**Notes for completeness:**
- When first turned on the selected Tax Solution should be the first one in the list. Afterward, it should be the one that was previously selected
- The tax solution ID should be saved in the config under `tax.taxSolutionID`:
![Image](https://github.com/user-attachments/assets/6d3f7dd8-1499-4c43-8b28-ff5db41e3bfb)

@allgandalf let me know if you have any questions! But we should be able to start taking proposals again now - to implement the solution described above.

Also tagging Design for their approval since I did these mocks independently of them.

**Comment by dannymcclain** [comment]
Mocks look good Yuwen!

Will enabling Tax in the Import options automatically tick on the Tax feature in More features? I would think it should, and then if you went to More features to view it, it would show as an enabled toggle that is locked (on) since the tax feature is being handled by the integration. Does that sound right to you?

**Comment by yuwenmemon** [comment]
Yes, it will after the taxes are imported from Sage Intacct!

**Comment by yuwenmemon** [comment]
Oops I realized the last header in one of my mocks says "Default vendor". That should say "Tax Solution" instead.

**Comment by yuwenmemon** [comment]
I'll update the API to add a new endpoint for setting the taxSolutionID. If someone turns tax off (i.e. enabled is "falsy") we should disable tax, but we can still preserve the saved taxSolutionID

**Comment by yuwenmemon** [comment]
The PR for the API is up. It will be called `UpdateSageIntacctTaxSolutionID,` and you just need to provide it with a `taxSolutionID`.

Side note: When you first enable tax, if no taxSolutionID is set I would just use the first value in the list.

**Comment by yuwenmemon** [comment]
The API should be available on staging now!

**Comment by yuwenmemon** [comment]
It's also now available on production

**Comment by garrettmknight** [comment]
@yuwenmemon / @allgandalf is there another PR we're waiting on here or is this good to test again?

**Comment by garrettmknight** [comment]
We've had a few PRs for this one. Can you confirm we're "done done"?

**Comment by garrettmknight** [comment]
**Payment Summary:**
- Contributor: @twilight2294 $250 via Upwork
- Contributor+: @allgandalf $250 via Upwork
